File: benchmark/2026-09-17-7677581(latest)/semantic-compression/probes/python/F05.P3.py
def probe() -> int:
    xs = [1, 2, 3]
    k = 0
    # BEGIN PROBE F05.P3
    def neg(a):
        return k - a

    ys = list(map(neg, xs))
    return ys[0]
    # END PROBE F05.P3


print(probe())

File: benchmark/2026-09-17-7677581(latest)/semantic-compression/probes/rust/F05.P3.rs
fn probe() -> i32 {
    let xs: Vec<i32> = vec![1, 2, 3];
    let k: i32 = 0;
// BEGIN PROBE F05.P3
let neg = |a: i32| k - a;
let ys: Vec<i32> = xs.into_iter().map(neg).collect();
ys[0]
// END PROBE F05.P3
}

fn main() {
    println!("{}", probe());
}
